netmap/netmap_contract.go

```diff
@@ -601,6 +610,40 @@ func ListConfig() []record {
 	return config
 }
 
+// SubscribeForNewEpoch registers passed contract as a NewEpoch event
+// subscriber. Such a contract must have a `NewEpoch` method with a single
+// numeric parameter. Transactions that call SubscribeForNewEpoch must be
+// witnessed by the Alphabet.
+// Produces `NewEpochSubscription` notification event with a just
+// registered recipient in a success case.
+func SubscribeForNewEpoch(contract interop.Hash160) {
+	common.CheckAlphabetWitness(common.AlphabetAddress())
+
+	if !management.HasMethod(contract, "newEpoch", 1) {
+		panic(address.FromHash160(contract) + " contract does not have `newEpoch(epoch)` method")
+	}
+
+	ctx := storage.GetContext()
+	var num byte
+	it := storage.Find(ctx, []byte(newEpochSubscribersPrefix), storage.KeysOnly|storage.RemovePrefix)
+	for iterator.Next(it) {
+		if common.BytesEqual(iterator.Value(it).(interop.Hash160), contract) {
```

Comment:
So, you're searching through `newEpochSubscribersPrefix`, removing prefix and they what you end up with is 21 bytes, index and hash. You need to strip index at least and then maybe use https://pkg.go.dev/github.com/nspcc-dev/neo-go/pkg/interop#Hash160.Equals

Reply:
> you end up with is 21 bytes

my bad, fixed

> maybe use https://pkg.go.dev/github.com/nspcc-dev/neo-go/pkg/interop#Hash160.Equals

ok, looks better